pages/product/index.vue

```diff
@@ -336,28 +352,35 @@ export default defineComponent({
       )
 
       if (isValid) {
-        setFulfillment(selectedFulfillmentVal, purchaseLocation.value.code)
+        setFulfillment(selectedFulfillmentVal, shortName, purchaseLocation?.value?.code)
+        checkEligibilityForAddToCart()
       }
     }
 
+    const checkEligibilityForAddToCart = () => {
+      isValidForAddToCart.value =
+        isProductVariationsSelected(product.value) && Boolean(product.value.fulfillmentMethod)
+    }
+
     // Add to Cart
     const quantityLeft = computed(() => 5)
     const qtySelected = useState(`pdp-selected-qty`, () => 1)
 
-    const addToCart = () => {
-      const addToCartVariables = {
-        productToAdd: {
-          product: {
-            productCode: product.value?.productCode,
-          },
-          quantity: qtySelected,
-          fulfillmentMethod: product.value?.fulfillmentMethod,
-          purchaseLocation: product.value?.purchaseLocationCode,
+    const addToCart = async () => {
+      const productToAdd: CartItem = {
```

Comment:
Use the AddToCartInput helper I added to create the productToAdd variable

Reply:
done